retroshare-gui/src/gui/NewsFeed.cpp

```diff
@@ -176,6 +181,88 @@ void NewsFeed::sortChanged(int index)
 	ui->feedWidget->setSortRole(ROLE_RECEIVED, sortOrder);
 }
 
+// handler for the new notification system in libretroshare.
+
+void NewsFeed::handleEvent(std::shared_ptr<const RsEvent> event)
+{
+	uint flags = Settings->getNewsFeedFlags();
+
+ 	const RsAuthSslConnectionAutenticationEvent *pssl_e = dynamic_cast<const RsAuthSslConnectionAutenticationEvent*>(event.get());
+
+    if(pssl_e != nullptr && (flags & (RS_FEED_TYPE_SECURITY | RS_FEED_TYPE_PEER)))
+    {
+        RsAuthSslConnectionAutenticationEvent e = *pssl_e;	// make a copy because we lose memory ownership here
+
+		RsQThreadUtils::postToObject( [=]()
+		{
+			/* Here it goes any code you want to be executed on the Qt Gui
+			 * thread, for example to update the data model with new information
+			 * after a blocking call to RetroShare API complete, note that
+			 * Qt::QueuedConnection is important!
+			 */
```

Comment:
You need a copy of the `shared_ptr` in this context otherwise the event is probably deleted before this function could be executed. So you should reference `event` directly inside the lambda body so the lambda will automatically `[=]` make a copy for you, so you should move also the cast inside the lambda and upstair instead of checking for `nullptr` in the cast check for `event->mType == RsEventType::AUTHSSL_CONNECTION_AUTENTICATION` which is also cheaper

Reply:
actually I just make the copy line 194. The original shared_ptr is not used in the function.

Reply:
the problem is that if you don' refer to a `shared_ptr` in that context the object will not survive, and when the function is executed in the Qt thread `e` will point to an already deleted memory location, so you need to refer to the original `shared_ptr` or a properly `dynamic_pointer_cast` copy of it and not to a raw pointer in the place you will finally use it